format/spec.md

```diff
@@ -607,6 +611,8 @@ Notes:
 
 1. An alternative, *strict projection*, creates a partition predicate that will match a file if all of the rows in the file must match the scan predicate. These projections are used to calculate the residual predicates for each file in a scan.
 2. For example, if `file_a` has rows with `id` between 1 and 10 and a delete file contains rows with `id` between 1 and 4, a scan for `id = 9` may ignore the delete file because none of the deletes can match a row that will be selected.
+3. Floating point partition values are considered equal if there IEEE 754 floating-point “single format” bit layout
```

Comment:
nit: 'if their IEEE'?

Reply:
Why do we need this? I don't think you should delete based on floating point values since there are so many ways for it to go wrong. Maybe we should disallow deleting by floating points instead of clarifying this?

Reply:
fixed typo.

@rdblue this is a separate clarification from https://github.com/apache/iceberg/pull/8981 for equality deletes floating points are already prohibited (as they are for a primary key).  For partitions it appears identity transforms are allowed to apply to floating point numbers, this tries to clarify how comparisons should be done in this scenario (I assume we can't exclude floating point numbers from identity transforms at this point?)

Reply:
No, I don't think that we can exclude floating point numbers from identity transforms. I think it's fine though, as long as equality is well understood.